src/librustc_lint/lib.rs

```diff
@@ -0,0 +1,137 @@
+pub use rustc::metadata as metadata;
+pub use rustc::middle as middle;
+pub use rustc::session as session;
+pub use rustc::util as util;
+
+use session::Session;
+use lint::{LintPassObject, LintId};
+
+mod builtin;
+
+/// Tell the `LintStore` about all the built-in lints (the ones
+/// defined in this crate and the ones defined in
+/// `rustc::lint::builtin`).
+pub fn register_builtins(store: &mut lint::LintStore, sess: Option<&Session>) {
+    macro_rules! add_builtin {
+        ($sess:ident, $($name:ident),*,) => (
+            {$(
+                store.register_pass($sess, false, box builtin::$name as LintPassObject);
+                )*}
+            )
+    }
+
+    macro_rules! add_builtin_with_new {
+        ($sess:ident, $($name:ident),*,) => (
+            {$(
+                store.register_pass($sess, false, box builtin::$name::new() as LintPassObject);
+                )*}
+            )
+    }
+
+    macro_rules! add_lint_group {
+        ($sess:ident, $name:expr, $($lint:ident),*) => (
+            store.register_group($sess, false, $name, vec![$(LintId::of(builtin::$lint)),*]);
+            )
+    }
+
+    add_builtin!(sess,
+                 HardwiredLints,
+                 WhileTrue,
+                 UnusedCasts,
+                 ImproperCTypes,
+                 BoxPointers,
+                 UnusedAttributes,
+                 PathStatements,
+                 UnusedResults,
+                 NonCamelCaseTypes,
+                 NonSnakeCase,
+                 NonUpperCaseGlobals,
+                 UnusedParens,
+                 UnusedImportBraces,
+                 NonShorthandFieldPatterns,
+                 UnusedUnsafe,
+                 UnsafeCode,
+                 UnusedMut,
+                 UnusedAllocation,
+                 MissingCopyImplementations,
+                 UnstableFeatures,
+                 Stability,
+                 UnconditionalRecursion,
+                 InvalidNoMangleItems,
+                 PluginAsLibrary,
+                 );
+
+    add_builtin_with_new!(sess,
+                          TypeLimits,
+                          RawPointerDerive,
+                          MissingDoc,
+                          MissingDebugImplementations,
+                          );
+
+    add_lint_group!(sess, "bad_style",
+                    NON_CAMEL_CASE_TYPES, NON_SNAKE_CASE, NON_UPPER_CASE_GLOBALS);
+
+    add_lint_group!(sess, "unused",
+                    UNUSED_IMPORTS, UNUSED_VARIABLES, UNUSED_ASSIGNMENTS, DEAD_CODE,
+                    UNUSED_MUT, UNREACHABLE_CODE, UNUSED_MUST_USE,
+                    UNUSED_UNSAFE, PATH_STATEMENTS);
+
+    // We have one lint pass defined specially
+    store.register_pass(sess, false, box lint::GatherNodeLevels as LintPassObject);
+
+    // Insert temporary renamings for a one-time deprecation
+    store.register_renamed("raw_pointer_deriving", "raw_pointer_derive");
+
+    store.register_renamed("unknown_features", "unused_features");
```

Comment:
I'm happy to remove them, but they're very low maintenance and it is helpful for people updating. *shrug*